nixos/modules/config/stub-ld.nix

```diff
@@ -0,0 +1,81 @@
+{ config, lib, pkgs, ... }:
+
+let
+  inherit (lib) last splitString getExe mkOption types mdDoc mkIf optionals;
+  inherit (builtins) elem toFile;
+
+  stub-ld = pkgs: let
+    message = ''
+      You seem to be trying to run a dynamically linked executable that
+      wasn't made for NixOS. This is possible, but not without extra steps,
+      because NixOS, by design, does not have a global library path.
+
+      There are several solutions available:
+
+      1. If the program is packaged in nixpkgs, use that version.
+
+      2. Write a nix expression for the program to package it
+         yourself. There are 3 approaches to this:
+
+         a. Build from source. This is ideal.
+
+         b. Fix the program's ELF metadata to hardcode paths to libraries
+            using autoPatchelfHook. Use this if building from source isn't
+            feasible.
+
+         c. Wrap the program to run in a normal-seeming container using
+            buildFHSEnv. This is undesirable, but sometimes necessary, for
+            example, if the package downloads and runs further executables.
+
+      3. If you expect to be doing this sort of thing a lot, you can set up
+         nix-ld using programs.nix-ld.enable. Make sure to add any needed
+         dependencies to programs.nix-ld.libraries.
+
+      4. Try running your program with steam-run, which uses the same
+         buildFHSEnv container as the steam package. This is quick and easy,
+         and usually works, but isn't a particularly good long-term solution.
+    '';
+
+    inherit (pkgs.stdenv) system;
+
+    libDir = if elem system [ "x86_64-linux" "mips64-linux" "powerpc64le-linux" ]
+             then "lib64"
+             else "lib";
```

Comment:
This should probably not be a list maintained here but rather a property of the respective systems.

Reply:
It is a somewhat odd property for nixpkgs to be keeping track of, because it's not a matter of how nix programs behave on those systems, just how *other* distros behave. Still, you may be right.

I'm not even sure how to add a property to the system attrset (I'm sure I can figure it out), or how to access it reliably if I did (less sure I can figure that out), though. Using `pkgs.hostPlatform` made ofborg fail to eval earlier, making me switch to `pkgs.stdenv.system`, even though I *think* that's the build system in cross-compiling cases, which isn't what I want. :-/

Reply:
I guess the way to access it would be through `pkgs.stdenv.hostPlatform`? `pkgs.hostPlatform` doesn't seem to be available in all cases, but hopefully that is.

Reply:
[`hostPlatform` is an alias to `stdenv.hostPlatform`](https://github.com/NixOS/nixpkgs/blob/86a8472db9c125fef4da159e7d0e90dbcb739873/pkgs/top-level/aliases.nix#L1030). Same with `system`. I think ofBorg was made to not like aliases, probably for good reason.

Reply:
Ah, that explains it.